### PR TITLE
Audit feature toggles

### PR DESCRIPTION
**Pull request by govindkrjoshi**
**Story:** https://www.pivotaltracker.com/story/show/159097919

### CODE REVIEW

File: doc/feature-toggles.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Feature toggles are used the application to turn on, or off certain features of the backend.
+
+There are two kinds of feature toggles used currently.
+  - Boolean Toggles - Used to turn on or off a feature entirely.
+    They are turned on by setting env variable `ENABLE_<capitalized_feature_name>` to true.
+    Boolean feature toggles are turned off by default
+  - Regex matchs: These are used to turn on or off a feature, by comparing strings to the provided regex
+    They are turned on by setting env variable `ENABLE_REGEX_MATCH_FOR_<capitalized_feature_name>` to the allowed regex.
+
+### Active Feature
```
Comment (timcheadle) [inline doc/feature-toggles.md]: Should be "Features" (plural)

File: doc/feature-toggles.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+#### SMS_NOTIFICATION_FOR_OTP
+Enables SMSes to be sent with OTPs for login
+The feature can be turned off in environments where otp sms notifications need not be sent, eg: qa, development
+
+#### ACCESSIBLE_SYNC_APIS
+This is the only example of a regex match feature toggle.
+Allows selectively enabling the sync apis. Sync apis which match the regex are turned on.
+
+### Deprecated Feature
```
Comment (timcheadle) [inline doc/feature-toggles.md]: These aren't deprecated; they're deleted. Let's delete this section so it isn't confusing.

File: doc/feature-toggles.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+## Feature Toggles
+
+Feature toggles are used the application to turn on, or off certain features of the backend.
+
+There are two kinds of feature toggles used currently.
```
Comment (timcheadle) [inline doc/feature-toggles.md]: This section is confusing. Let's just document each flag specifically, including whether it's a boolean or a regex, and what the default is. Redundant documentation is easier to read.

File: doc/feature-toggles.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  - Boolean Toggles - Used to turn on or off a feature entirely.
+    They are turned on by setting env variable `ENABLE_<capitalized_feature_name>` to true.
+    Boolean feature toggles are turned off by default
+  - Regex matchs: These are used to turn on or off a feature, by comparing strings to the provided regex
+    They are turned on by setting env variable `ENABLE_REGEX_MATCH_FOR_<capitalized_feature_name>` to the allowed regex.
+
+### Active Feature
+
+#### PURGE_ENDPOINT_FOR_QA
```
Comment (timcheadle) [inline doc/feature-toggles.md]: Should be `ENABLE_PURGE_ENDPOINT_FOR_QA`, and list its default value

File: doc/feature-toggles.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+Allows the database to be cleared up after running integration tests.
+This feature should only be turned on for the QA env.
+
+#### SMS_NOTIFICATION_FOR_OTP
```
Comment (timcheadle) [inline doc/feature-toggles.md]: Should be `ENABLE_SMS_NOTIFICATION_FOR_OTP`, and list its default value

File: doc/feature-toggles.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+Enables SMSes to be sent with OTPs for login
+The feature can be turned off in environments where otp sms notifications need not be sent, eg: qa, development
+
+#### ACCESSIBLE_SYNC_APIS
```
Comment (timcheadle) [inline doc/feature-toggles.md]: Should be `ENABLE_REGEX_MATCH_FOR_ACCESSIBLE_SYNC_APIS`, and list its default value

Also, this flag name is so long it feems like a code smell. Maybe we can change it to `ENABLE_MATCHING_SYNC_APIS`?

File: doc/feature-toggles.md (hunk 3 above)
Comment (govindkrjoshi) [reply doc/feature-toggles.md]: 👍

File: doc/feature-toggles.md (hunk 1 above)
Comment (govindkrjoshi) [reply doc/feature-toggles.md]: Fixed.

File: doc/feature-toggles.md (hunk 2 above)
Comment (govindkrjoshi) [reply doc/feature-toggles.md]: Removed this section.

File: doc/feature-toggles.md (hunk 4 above)
Comment (govindkrjoshi) [reply doc/feature-toggles.md]: Fixed.

File: doc/feature-toggles.md (hunk 5 above)
Comment (govindkrjoshi) [reply doc/feature-toggles.md]: Fixed.

File: doc/feature-toggles.md (hunk 6 above)
Comment (govindkrjoshi) [reply doc/feature-toggles.md]: `ENABLE_REGEX_MATCH_FOR_` set as a prefix to distinguish from boolean feature toggle. 
Agreed that the name is long and awkward. Changed it to `ENABLE_REGEX_MATCHING_SYNC_APIS`

**Pull request merged**